src/components/settings/settings/EditSettingsForm.tsx

```diff
@@ -384,6 +384,14 @@ class EditSettingsForm extends Component<IProps, IState> {
       e.preventDefault();
     }
 
+    // Do not submit if the accent color is not set
+    if (
+      this.props.form.$('accentColor').value === '#' ||
```

Comment:
Why not use the `isValidColor` you implemented in `color-helpers.ts` both for `accentColor` and for `progressbarAccentColor` directly here? And then remove the tests on lines 400 and 404?

Otherwise, we could do this inside the `form.submit` function below, and change the current lines 400-402 as follows:
```
if (accentColor.trim().length === 0 || isValidColor(accentColor)) {
    values.accentColor = DEFAULT_ACCENT_COLOR;
}
```
and similarly for `progressbarAccentColor` on lines 404-406. That way we fully verify the user input. 

What do you think?

Reply:
I've initially used that but I had to prevent the submit somehow - when the user is typing every keystroke submits to the file (I know this is not ideal, but it is how it is coded.... we would need to redo the entire logic to improve this at some point). So, in order to allow the user to input the hex, I can't tell it to submit "#", so I just prevent the submit entirely. It's a bit of a workaround.

But if you have any idea on how to code it better I would be up for improving this xD I just didn't gave it much thought myself as it was taking more time from me as it was supposed to (given the fact that is a simple issue --')